### PR TITLE
support skipEncryptRewrite hint to force skip encrypt rewrite

### PR DESCRIPTION
**Pull request by cheese8**
Ref #12957.

SS already support null, and here is hint way. @strongduanmu 

Changes proposed in this pull request:
-support set blank to clear plain after encrypt

### CODE REVIEW

**Comment by TeslaCN** [comment]
Hi @strongduanmu 
Could you take a look at this?

**Comment by strongduanmu** [comment]
> #12957

@TeslaCN This PR needs more discussion, so it will be better for the next release.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/hint/SQLHintExtractor.java [hunk 1]
```diff
@@ -66,4 +66,13 @@ public Optional<String> findHintDataSourceName() {
...
         return sqlHintProperties.getValue(SQLHintPropertiesKey.WRITE_ROUTE_ONLY_KEY);
     }
+    
+    /**
+     * Judge whether is hint indicated encrypt rewrite needed or not.
+     *
+     * @return whether is hint indicated encrypt rewrite needed or not
+     */
+    public boolean isHintEncryptRewriteNeeded() {
+        return sqlHintProperties.getValue(SQLHintPropertiesKey.NO_ENCRYPT_REWRITE_KEY);
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/hint/SQLHintExtractor.java:76]: @cheese8 Can you rename NO_ENCRYPT_REWRITE_KEY to SKIP_ENCRYPT_REWRITE_KEY?

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/hint/SQLHintPropertiesKey.java [hunk 2]
```diff
@@ -41,7 +41,12 @@ public enum SQLHintPropertiesKey implements TypedPropertyKey {
     /**
      * Whether to use traffic or not.
      */
-    USE_TRAFFIC("useTraffic", String.valueOf(Boolean.FALSE), boolean.class);
+    USE_TRAFFIC("useTraffic", String.valueOf(Boolean.FALSE), boolean.class),
+    
+    /**
+     * Whether hint indicate encrypt rewrite needed or not.
+     */
+    NO_ENCRYPT_REWRITE_KEY("noEncryptRewrite", String.valueOf(Boolean.FALSE), boolean.class);
```
Comment (strongduanmu) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/hint/SQLHintPropertiesKey.java:49]: @cheese8 Can you rename noEncryptRewrite to skipEncryptRewrite?

File: shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/query-with-cipher/dml/update/update.xml [hunk 3]
```diff
@@ -21,31 +21,51 @@
         <input sql="UPDATE t_account SET account_id = ?, certificate_number = ?, password = ?, amount = ?, status = ? WHERE account_id = ? AND certificate_number = ? AND password = ? AND amount = ? AND status = ?" parameters="1, 111X, aaa, 1000, OK, 1, 111X, aaa, 1000, OK" />
         <output sql="UPDATE t_account SET account_id = ?, cipher_certificate_number = ?, assisted_query_certificate_number = ?, cipher_password = ?, assisted_query_password = ?, cipher_amount = ?, status = ? WHERE account_id = ? AND assisted_query_certificate_number = ? AND assisted_query_password = ? AND cipher_amount = ? AND status = ?" parameters="1, encrypt_111X, assisted_query_111X, encrypt_aaa, assisted_query_aaa, encrypt_1000, OK, 1, assisted_query_111X, assisted_query_aaa, encrypt_1000, OK" />
     </rewrite-assertion>
-
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/query-with-cipher/dml/update/update.xml:24]: Please keep same indent with previous line.

File: shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/query-with-cipher/dml/update/update.xml [hunk 4]
```diff
@@ -21,31 +21,51 @@
...
     </rewrite-assertion>
-    
+
     <rewrite-assertion id="update_for_literals" db-types="MySQL">
         <input sql="UPDATE t_account SET account_id = 1, certificate_number = '111X', password = 'aaa', amount = 1000, status = 'OK' WHERE account_id = 1 AND certificate_number = '111X' AND password = 'aaa' AND amount = 1000 AND status = 'OK'" />
         <output sql="UPDATE t_account SET account_id = 1, cipher_certificate_number = 'encrypt_111X', assisted_query_certificate_number = 'assisted_query_111X', cipher_password = 'encrypt_aaa', assisted_query_password = 'assisted_query_aaa', cipher_amount = 'encrypt_1000', status = 'OK' WHERE account_id = 1 AND assisted_query_certificate_number = 'assisted_query_111X' AND assisted_query_password = 'assisted_query_aaa' AND cipher_amount = 'encrypt_1000' AND status = 'OK'" />
     </rewrite-assertion>
-
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/query-with-cipher/dml/update/update.xml:29]: Please keep same indent with previous line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/query-with-cipher/dml/update/update.xml (hunk 4 above)
Comment (cheese8) [reply shardingsphere-test/shardingsphere-rewrite-test/src/test/resources/scenario/encrypt/case/query-with-cipher/dml/update/update.xml:29]: ok

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**